Packs/AzureActiveDirectory/Integrations/AzureADIdentityProtection/AzureADIdentityProtection.py

```diff
@@ -329,32 +383,32 @@ def date_str_to_azure_format(date_str):
 
 
 def fetch_incidents(client: AADClient, params: Dict[str, str]):
-    last_run: Dict[str, str] = demisto.getLastRun()
-    demisto.debug(f'[AzureIdentityProtection] last run: {last_run}')
-
-    last_fetch, last_fetch_datetime = get_last_fetch_time(last_run, params)
-    query_filter = build_filter(last_fetch, params)
-    demisto.debug(f'[AzureIdentityProtection] last fetch is: {last_fetch}, filter is: {query_filter}')
-
-    risk_detection_list_raw: Dict = client.azure_ad_identity_protection_risk_detection_list_raw(
-        limit=int(params.get('max_fetch', '50')),
-        filter_expression=query_filter,
-        user_id=params.get('fetch_user_id', ''),
-        user_principal_name=params.get('fetch_user_principal_name', ''),
-    )
+    def fetch_func(last_fetch):
+        query_filter = build_filter(last_fetch, params)
+        demisto.debug(f'[AzureIdentityProtection] last fetch is: {last_fetch}, filter is: {query_filter}')
+
+        risk_detection_list_raw: Dict = client.azure_ad_identity_protection_risk_detection_list_raw(
+            limit=int(params.get('max_fetch', '50')),
+            filter_expression=query_filter,
+            user_id=params.get('fetch_user_id', ''),
+            user_principal_name=params.get('fetch_user_principal_name', ''),
+        )
 
-    detections: list = risk_detection_list_raw.get('value', [])
+        detections: list = risk_detection_list_raw.get('value', [])
 
-    incidents, latest_detection_time = detections_to_incidents(detections, last_fetch_datetime=last_fetch_datetime)
-    demisto.debug(f'[AzureIdentityProtection] Fetched {len(incidents)} incidents')
+        last_fetch_datetime: datetime = datetime.strptime(last_fetch, DATE_FORMAT)
+        incidents, latest_detection_time = detections_to_incidents(detections, last_fetch_datetime=last_fetch_datetime)
+        demisto.debug(f'[AzureIdentityProtection] Fetched {len(incidents)} incidents')
+        latest_detection_time = latest_detection_time.strftime(DATE_FORMAT)
+        return {
+            'last_time_run': latest_detection_time
```

Comment:
Missing last_etag?

Reply:
You don't use etag so no need to put it.